src/Microsoft.AspNet.Mvc.Abstractions/ModelBinding/ModelBindingResult.cs

```diff
@@ -9,6 +9,16 @@ namespace Microsoft.AspNet.Mvc.ModelBinding
     public class ModelBindingResult
     {
         /// <summary>
+        /// Creates a new <see cref="ModelBindingResult"/> indicating a fatal error.
+        /// </summary>
+        /// <param name="key">The key using which was used to attempt binding the model.</param>
+        public ModelBindingResult(string key)
```

Comment:
@rynowak if you feel strongly about the factory method approach, please file a bug. I hesitate because this would affect all `IModelBinder` implementations.